crates/sui-adapter/src/execution_engine.rs

```diff
@@ -226,18 +226,23 @@ fn execute_transaction<
         };
         execution_result
     });
-    if !gas_status.is_unmetered() {
-        temporary_store.charge_gas(gas_object_ref.0, &mut gas_status, &mut result, gas);
-    }
-    if !is_system {
-        #[cfg(debug_assertions)]
-        {
-            if !Mode::allow_arbitrary_values() {
-                // ensure that this transaction did not create or destroy SUI
-                temporary_store.check_sui_conserved().unwrap();
+    #[allow(clippy::collapsible_if)]
+    if protocol_config.gas_model_version() > 1 {
+        if !is_system {
+            temporary_store.charge_gas(gas_object_ref.0, &mut gas_status, &mut result, gas);
+            if protocol_config.conservation_checks() {
```

Comment:
```suggestion
            if protocol_config.conservation_checks() || cfg!(debug_assertions) {
```

I'd like to make sure these always run in tests/simtests even if they're off in the protocol config